### PR TITLE
Fix nre issue #20

### PR DESCRIPTION
**Pull request by flaviut**
Fix flaviut/nre#20.

### CODE REVIEW

File: lib/impure/nre.nim [hunk 1]
```diff
@@ -513,7 +513,8 @@ iterator findIter*(str: string, pattern: Regex, start = 0, endpos = int.high): R
   let matchesCrLf = pattern.matchesCrLf()
   let unicode = uint32(getinfo[culong](pattern, pcre.INFO_OPTIONS) and
     pcre.UTF8) > 0u32
-  let strlen = if endpos == int.high: str.len else: endpos+1
+  # TODO cast[int] in order to work around bugs in ranges
+  let strlen = if endpos == int.high: str.len else: cast[int](endpos) + 1
```
Comment (Araq) [inline lib/impure/nre.nim]: Er ... let's better fix the codegen instead.

**Comment by Araq** [comment]
Works for me without this 'cast' workaround, but there has been a recent regression with integer ops which you might have run into? Test again against devel please.

**Comment by flaviut** [comment]
I've pulled the latest devel, `koch boot`'d, but it fails to compile without that change:

```
❯❯❯ nim c -r tests/stdlib/tnre.nim
<snip>
Hint: used config file '/home/user/dev/nim/nimlang/config/nim.cfg' [Conf]
Hint: system [Processing]
Hint: tnre [Processing]
Hint: nre [Processing]
Hint: pcre [Processing]
Hint: util [Processing]
Hint: tables [Processing]
Hint: hashes [Processing]
Hint: strutils [Processing]
Hint: parseutils [Processing]
Hint: etcpriv [Processing]
Hint: math [Processing]
Hint: times [Processing]
Hint: unsigned [Processing]
Hint: options [Processing]
Hint: typetraits [Processing]
Hint: unicode [Processing]
lib/impure/nre.nim(591, 3) template/generic instantiation from here
lib/impure/nre.nim(516, 59) Error: over- or underflow
```

**Comment by Araq** [comment]
Anyway, get rid of this workaround please. The overflow is a separate issue.

**Comment by flaviut** [comment]
Done.

**Pull request merged**